File: 122A/main.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    int n;
    cin >> n;
    bool flag = true;
    string s = to_string(n);
    
    for (char c : s) {
        if (c != '4' && c != '7') {
            flag = false;
            break;
        }
    }
    
    if (flag) {
        cout << "YES";
        return 0;
    }
    
    bool almost = false;
    for (int i = 1; i <= n; i++) {
        bool flag_i = true;
        string num = to_string(i);
        
        for (char c : num) {
            if (c != '4' && c != '7') {
                flag_i = false;
                break;
            }
        }
        
        if (flag_i && n % i == 0) {
            almost = true;
            break;
        }
    }
    
    cout << (almost ? "YES" : "NO");
    return 0;
}

File: 122A/main.py
n = int(input())
flag = True

for digit in str(n):
    if digit not in {'4', '7'}:
        flag = False
        break

if flag:
    print("YES")
else:
    almost = False
    for i in range(1, n + 1):
        flag_i = True
        for digit in str(i):
            if digit not in {'4', '7'}:
                flag_i = False
                break
        
        if flag_i and n % i == 0:
            almost = True
            break 
    
    print("YES" if almost else "NO")